tools/helloworld/actions/GreetingAction__.php

```diff
@@ -4,7 +4,7 @@
 
 class GreetingAction__ extends YesWikiAction
 {
-    function run($parent) 
+    function run()
```

Comment:
Donc grosso modo tu supprimes la méthode "clean" proposée par @seballot (avec, notamment, la possibilité de passer des arguments à des "after callbacks" via $parent) pour la remplacer par une méthode non-clean, juste dans le but de le rendre rétro-compatible ? Pour moi c'est comme annuler le refactoring :-/

Reply:
Quand on travaille avec des classes, cela ne change rien. Tu pouvais avant mes modifs passer des arguments via $parent->arguments.

Là où le code a évolué, c'est qu'on a maintenant accès aux arguments via $this->arguments qu'on soit dans les callback ou la classe parent. Cela a permis de simplifier le code et l'a rendu à mon sens un peu plus propre : `abstract public function run($arguments);` de YesWikiPerformable était instanciée avec des objets différents : $arguments pour la classe parent et $parent quand c'était un callback. Maintenant, l'utilisateur n'a plus besoin de faire attention à cette spécificité et on a un seul accès aux arguments via $this->arguments.